src/pools/managedPools/MaganedPoolKassandra.ts

```diff
@@ -0,0 +1,285 @@
+import { BigNumber, formatFixed, parseFixed } from '@ethersproject/bignumber';
+import { WeiPerEther as ONE, Zero } from '@ethersproject/constants';
+import { getAddress } from '@ethersproject/address';
+import {
+    BigNumber as OldBigNumber,
+    ZERO,
+    bnum,
+    scale,
+} from '../../utils/bignumber';
+import { isSameAddress } from '../../utils';
+import {
+    PoolBase,
+    PoolPairBase,
+    PoolTypes,
+    SubgraphPoolBase,
+    SwapTypes,
+} from '../../types';
+import { WeightedPoolToken } from '../weightedPool/weightedPool';
+import {
+    _calcInGivenOut,
+    _calcOutGivenIn,
+    _derivativeSpotPriceAfterSwapExactTokenInForTokenOut,
+    _derivativeSpotPriceAfterSwapTokenInForExactTokenOut,
+    _spotPriceAfterSwapExactTokenInForTokenOut,
+    _spotPriceAfterSwapTokenInForExactTokenOut,
+} from '../weightedPool/weightedMath';
+import { universalNormalizedLiquidity } from '../liquidity';
+
+export type ManagedPoolPairData = PoolPairBase & {
+    weightIn: BigNumber;
+    weightOut: BigNumber;
+};
+
+export class MaganedPoolKassandra implements PoolBase<PoolPairBase> {
+    poolType: PoolTypes = PoolTypes.Managed;
+    id: string;
+    address: string;
+    tokensList: string[];
+    tokens: WeightedPoolToken[];
+    totalWeight: BigNumber;
+    totalShares: BigNumber;
+    swapFee: BigNumber;
+    MAX_IN_RATIO = parseFixed('0.3', 18);
+    MAX_OUT_RATIO = parseFixed('0.3', 18);
+    mainIndex?: number | undefined;
+    isLBP?: boolean | undefined;
+
+    constructor(
+        id: string,
+        address: string,
+        tokenList: string[],
+        tokens: WeightedPoolToken[],
+        totalShares: string,
+        swapFee: string
+    ) {
+        this.id = id;
+        this.address = address;
+        this.tokensList = tokenList;
+        this.tokens = tokens;
+        this.totalShares = parseFixed(totalShares, 18);
+        this.totalWeight = parseFixed('1', 18);
+        this.swapFee = parseFixed(swapFee, 18);
+    }
+
+    static fromPool(pool: SubgraphPoolBase): MaganedPoolKassandra {
+        if (!pool.totalWeight) {
+            throw new Error('WeightedPool missing totalWeight');
+        }
```

Comment:
Since you're considering totalWeight as always 1, I'd say it's safer to remove this check, otherwise an eventual issue with SG may break this flow.
```suggestion
```